src/ios/BEMShakeNotifier.m

```diff
@@ -0,0 +1,431 @@
+//
+//  BEMShakeNotifier.m
+//  emission
+//
+//  Created by Felipe Arias
+//
+//
+#import <CoreMotion/CoreMotion.h>
+#import <math.h>
+#import <Foundation/Foundation.h>
+#import <QuartzCore/QuartzCore.h>
+#import "BEMShakeNotifier.h"
+#import "TripDiaryStateMachine.h"
+#import "APPLocalNotification.h"
+#import "BEMBuiltinUserCache.h"
+#import "DataUtils.h"
+#import "LocalNotificationManager.h"
+#import "SimpleLocation.h"
+#import "PotentialIncident.h"
+#import "Transition.h"
+#import "MotionActivity.h"
+
+
+#define TRIP_STARTED @"trip_started"
+#define TRIP_ENDED @"trip_ended"
+#define TRACKING_STARTED @"tracking_started"
+#define TRACKING_STOPPED @"tracking_stopped"
+#define POTENTIAL_INCIDENT @"potential_incident"
+
+#define CONFIG_LIST_KEY @"config_list"
+#define MUTED_LIST_KEY @"muted_list"
+
+#define isNSNull(value) [value isKindOfClass:[NSNull class]]
+
+@interface BEMShakeNotifier () {
+    // Member variables go here.
+}
+
+@property (nonatomic,strong) NSMutableDictionary *observerMap;
+- (void)pluginInitialize;
+- (void)fireGenericTransitionFor:(NSString*) transition withUserInfo:(NSDictionary*) userInfo;
+- (void)notifyEvent:(NSString *)eventName data:(NSDictionary*)data;
+- (void)addEventListener:(CDVInvokedUrlCommand*)command;
+- (void)removeEventListener:(CDVInvokedUrlCommand*)command;
+- (void)enableSensor;
+- (void)disableSensor;
+
+@end
+
+@implementation BEMShakeNotifier
+
+
+- (void)pluginInitialize
+{
+    // TODO: We should consider adding a create statement to the init, similar
+    // to android - then it doesn't matter if the pre-populated database is not
+    // copied over.
+    NSLog(@"BEMShakeNotifier:pluginInitialize singleton -> initialize statemachine and delegate");
+    __typeof(self) __weak weakSelf = self;
+    [[NSNotificationCenter defaultCenter] addObserverForName:CFCTransitionNotificationName object:nil queue:nil
```

Comment:
why do you need this? iOS piggybacks on the location changes, so will only sense when the location tracking is on.